ephios/core/views/event.py

```diff
@@ -232,14 +233,25 @@ def get_queryset(self):
                 },
             )
             .select_related("type")
-            .prefetch_related("shifts")
-            .prefetch_related(Prefetch("shifts__participations"))
         )
         if self.filter_form.is_valid():
             qs = self.filter_form.filter_events(qs)
         else:
             # safeguard for not loading too many events
             qs = qs.filter(end_time__gte=timezone.now()).order_by("start_time", "end_time")
+        qs = qs.prefetch_related("shifts__participations")
+
+        # annotate groups that can view the event by prefetching the object permission model
+        qs = qs.prefetch_related(
+            Prefetch(
+                "group_object_permission_set",
+                queryset=GroupObjectPermission.objects.filter(permission__codename="view_event"),
```

Comment:
this includes all group with change permissons, right? probably related to #1258

Reply:
It does, yes.